fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -206,22 +302,37 @@ ACTOR Future<Void> rocksDBMetricLogger(std::shared_ptr<rocksdb::Statistics> stat
 		{ "BaseLevel", rocksdb::DB::Properties::kBaseLevel },
 		{ "EstPendCompactBytes", rocksdb::DB::Properties::kEstimatePendingCompactionBytes },
 	};
+
+	state std::unordered_map<std::string, uint64_t> readIteratorPoolStats = {
+		{ "NumReadIteratorsCreated", 0 },
+		{ "NumTimesReadIteratorsReused", 0 },
+	};
+
 	loop {
 		wait(delay(SERVER_KNOBS->ROCKSDB_METRICS_DELAY));
 		TraceEvent e("RocksDBMetrics");
+		uint64_t stat;
 		for (auto& t : tickerStats) {
 			auto& [name, ticker, cum] = t;
-			uint64_t val = statistics->getTickerCount(ticker);
-			e.detail(name, val - cum);
-			cum = val;
+			stat = statistics->getTickerCount(ticker);
+			e.detail(name, stat - cum);
+			cum = stat;
 		}
 
 		for (auto& p : propertyStats) {
 			auto& [name, property] = p;
-			uint64_t stat = 0;
+			stat = 0;
 			ASSERT(db->GetIntProperty(property, &stat));
 			e.detail(name, stat);
 		}
+
+		stat = readIterPool->numReadIteratorsCreated();
```

Comment:
Can you use a for-loop to iterate the Iterator related metrics array? similar to propertyStats and tickerStats